msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IBroker.java

```diff
@@ -3,58 +3,84 @@
 
 package com.microsoft.aad.msal4j;
 
-import java.util.Set;
+import com.nimbusds.jwt.JWTParser;
+
+import java.net.URL;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 /**
  * Used to define the basic set of methods that all Brokers must implement
  *
- * All methods are so they can be referenced by MSAL Java without an implementation, and by default simply throw an
- * exception saying that a broker implementation is missing
+ * All methods are marked as default so they can be referenced by MSAL Java without an implementation,
+ *  and most will simply throw an exception if not overridden by an IBroker implementation
  */
 public interface IBroker {
 
     /**
-     * checks if a IBroker implementation exists
+     * Checks if an IBroker implementation is accessible by MSAL Java
      */
-
     default boolean isAvailable(){
         return false;
     }
+
     /**
      * Acquire a token silently, i.e. without direct user interaction
      *
      * This may be accomplished by returning tokens from a token cache, using cached refresh tokens to get new tokens,
      * or via any authentication flow where a user is not prompted to enter credentials
-     *
-     * @param requestParameters MsalRequest object which contains everything needed for the broker implementation to make a request
-     * @return IBroker implementations will return an AuthenticationResult object
      */
-    default IAuthenticationResult acquireToken(PublicClientApplication application, SilentParameters requestParameters) {
+    default void acquireToken(PublicClientApplication application, SilentParameters requestParameters, CompletableFuture<IAuthenticationResult> future) {
```

Comment:
Should not the methods' return type be CompletableFuture<IAuthenticationResult>? Keeping the return type as void will make these apis different from the MSAL Java acquireToken apis.

Reply:
I originally had it returning a CompletableFuture, however that made the code in MSAL Java itself kind of messy. If you look at `MsalRuntimeBroker`, each of the long `.thenApply(...);` sections would need to move into PublicClientApplication (or somewhere else in MSAL Java).

Doing that would also mean splitting up where the interop's result is retrieved and where it is parsed: rather than the IBroker implementation handling that entirely, it would be up to MSAL Java to call the correct methods to translate whatever the broker returned into an IAuthenticationResult